internal/service/advancedcluster/model_advanced_cluster_test.go

```diff
@@ -21,6 +23,128 @@ var (
 	advancedClusters = []admin.AdvancedClusterDescription{{StateName: conversion.StringPtr("NOT IDLE")}}
 )
 
+func TestFlattenReplicationSpecs(t *testing.T) {
+	var (
+		regionName         = "EU_WEST_1"
+		providerName       = "AWS"
+		expectedID         = "id1"
+		unexpectedID       = "id2"
+		expectedZoneName   = "z1"
+		unexpectedZoneName = "z2"
+		regionConfigAdmin  = []admin.CloudRegionConfig{{
+			ProviderName: &providerName,
+			RegionName:   &regionName,
+		}}
+		regionConfigTf = map[string]any{
+			"provider_name": "AWS",
+			"region_name":   regionName,
+		}
+		regionConfigTfDiffZone = map[string]any{
+			"provider_name": "AWS",
+			"region_name":   regionName,
+			"zone_name":     unexpectedZoneName,
+		}
+		admin1     = admin.ReplicationSpec{Id: &expectedID, ZoneName: &expectedZoneName, RegionConfigs: &regionConfigAdmin}
+		admin2     = admin.ReplicationSpec{Id: &unexpectedID, ZoneName: &unexpectedZoneName, RegionConfigs: &regionConfigAdmin}
+		testSchema = map[string]*schema.Schema{
+			"project_id": {Type: schema.TypeString},
+		}
+		tf1SameIDSameZone = map[string]any{
+			"id":             expectedID,
+			"num_shards":     1,
+			"region_configs": []any{regionConfigTf},
+			"zone_name":      expectedZoneName,
+		}
+		tf2NoIDSameZone = map[string]any{
+			"id":             nil,
+			"num_shards":     1,
+			"region_configs": []any{regionConfigTf},
+			"zone_name":      expectedZoneName,
+		}
+		tf3NoIDDiffZone = map[string]any{
+			"id":             nil,
+			"num_shards":     1,
+			"region_configs": []any{regionConfigTfDiffZone},
+			"zone_name":      unexpectedZoneName,
+		}
+		tf4diffIDDiffZone = map[string]any{
```

Comment:
curious on the numeration for these objects, is it needed or helps in some way?

Reply:
Removed in 9425ff49b8b3e0c3b0dad38fc44160d40e88033b